PWGJE/Tasks/jetHadronRecoil_OO.cxx

```diff
@@ -0,0 +1,541 @@
+                           filtered_Jets_PartLevel const& jets,
+                           aod::JetParticles const& particles)
+  {
+    spectra.fill(HIST("vertexZ"), collision.posZ());
+    fillMCPHistograms(jets, particles);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processMC_PartLevel, "process MC particle level", false);
+
+  void processMC_PartLevel_Weighted(filtered_Coll_PartLevel const& collision,
+                                    filtered_Jets_PartLevel const& jets,
+                                    aod::JetParticles const& particles)
+  {
+    auto weight = collision.weight();
+    spectra.fill(HIST("vertexZ"), collision.posZ(), weight);
+    fillMCPHistograms(jets, particles, weight);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processMC_PartLevel_Weighted, "process MC particle level with event weight", false);
+
+  void processJetsMatched(filtered_Coll_DetLevel_to_GetWeight const& collision,
+                          aod::JetMcCollisions const&,
+                          filtered_Tracks const& tracks,
+                          filtered_MatchedJets_DetLevel const& mcdjets,
+                          filtered_MatchedJets_PartLevel const& mcpjets)
+  {
+    if (skipEvent(collision))
+      return;
+    auto mcpjetsPerMCCollision = mcpjets.sliceBy(PartJetsPerCollision, collision.mcCollisionId());
+    fillMatchedHistograms(tracks, mcdjets, mcpjetsPerMCCollision);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processJetsMatched, "process matching of MC jets (no weight)", false);
+
+  void processJetsMatched_Weighted(filtered_Coll_DetLevel_to_GetWeight const& collision,
+                                   aod::JetMcCollisions const&,
+                                   filtered_Tracks const& tracks,
+                                   filtered_MatchedJets_DetLevel const& mcdjets,
+                                   filtered_MatchedJets_PartLevel const& mcpjets)
+  {
+    if (skipEvent(collision))
+      return;
+
+    auto mcpjetsPerMCCollision = mcpjets.sliceBy(PartJetsPerCollision, collision.mcCollisionId());
+    auto weight = collision.mcCollision().weight();
+
+    fillMatchedHistograms(tracks, mcdjets, mcpjetsPerMCCollision, weight);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processJetsMatched_Weighted, "process matching of MC jets (weighted)", false);
+
+  //------------------------------------------------------------------------------
+  // Auxiliary functions
+  template <typename Collision>
+  bool skipEvent(const Collision& coll)
+  {
+    /// \TODO: Do need !jetderiveddatautilities::selectTrigger(collision, triggerMaskBits) flag to skip an event?
+    return !coll.selection_bit(aod::evsel::kNoTimeFrameBorder) || !jetderiveddatautilities::selectCollision(coll, eventSelection);
+  }
+
+  template <typename Jet>
+  std::tuple<double, bool> isRecoilJet(const Jet& jet,
+                                       double phi_TT)
+  {
+    double dphi = fabs(TVector2::Phi_mpi_pi(jet.phi() - phi_TT));
```

Comment:
for this use the RecoDecay::constrainAngle function

Reply:
Added
Interesting to know why we should use ``RecoDecay::constrainAngle`` instead of ``TVector2::Phi_mpi_pi``?

Reply:
recodecay is the official place where these operations are done so the recommendation from O2 is to use that where possible. @vkucera can comment more

Reply:
@KotliarovAr O2 is C++ code, not ROOT code. There is no reason to use ROOT features if equivalents exist in standard C++ or O2.